pontoon/sync/tasks.py

```diff
@@ -283,34 +289,13 @@ def sync_translations(
                 if not vcs_project.synced_locales:
                     vcs_project.resources
 
-                if not obsolete_vcs_resources:
```

Comment:
This check is actually no longer needed. It's role was to prevent skipping the commit when files were removed from source repository - so we could also remove them from locale repository. But we no longer do that.

And, we should rename `obsolete_vcs_resources` to `removed_paths` in a followup.